kaggle_environments/core.py

```diff
@@ -533,7 +533,11 @@ def __run_interpreter(self, state):
                     args = [structify(state), self]
                     new_state = structify(self.interpreter(
                         *args[:self.interpreter.__code__.co_argcount]))
-                    for agent in new_state:
+                    for index, agent in enumerate(new_state):
+                        if index < len(log):
+                            duration = log[index]["duration"]
+                            overage_time_consumed = max(0, duration - self.configuration.actTimeout)
+                            agent.observation.remainingOverageTime -= overage_time_consumed
```

Comment:
This overage time update occurs after the environment interpreter updates its state -- there's no practical consequence for competitors but it might affect environment authors if they want to be able to control player overage time (e.g. to penalize a player for foul play).

Reply:
nit: I find it slightly awkward that we're updating here and then checking / erroring in `agent.py`, and relying on the shared logic matching up.  But perhaps it's not easy to combine them, or even based on your PR comment it's intentional (maybe should be a code comment?)

Reply:
I agree, I tried to find a nice way to implement this all in one location but it seemed like it was going to take a significant refactor to do well and I didn't want to dive into that quite yet. I'm angling for a big refactor project some time next year though that will decouple a lot of this code.